File: application/src/main/java/de/pbauerochse/worklogviewer/domain/timerangeprovider/CurrentMonthTimerangeProvider.java
package de.pbauerochse.worklogviewer.domain.timerangeprovider;

import de.pbauerochse.worklogviewer.domain.ReportTimerange;
import de.pbauerochse.worklogviewer.report.TimeRange;

/**
 * @author Patrick Bauerochse
 * @since 13.04.15
 */
public class CurrentMonthTimerangeProvider extends BaseTimerangeProvider {

    @Override
    public TimeRange getTimeRange() {
        return TimeRange.currentMonth();
    }

    @Override
    public ReportTimerange getReportTimerange() {
        return ReportTimerange.THIS_MONTH;
    }

}

File: application/src/main/java/de/pbauerochse/worklogviewer/timerange/CurrentMonthTimerangeProvider.kt
package de.pbauerochse.worklogviewer.timerange

import de.pbauerochse.worklogviewer.report.TimeRange

import de.pbauerochse.worklogviewer.util.FormattingUtil.getFormatted
import java.time.LocalDate

/**
 * TimeRange provider for the currrent month (including 1st till end of the month)
 */
internal object CurrentMonthTimerangeProvider : TimerangeProvider {
    override val label: String = getFormatted("timerange.thismonth")
    override val settingsKey: String = "THIS_MONTH"
    override val isComputed: Boolean = true
    override fun buildTimeRange(start: LocalDate?, end: LocalDate?): TimeRange {
        return TimeRange.currentMonth()
    }
}
